db/db_impl/db_impl_open.cc

```diff
@@ -1203,13 +1203,18 @@ Status DBImpl::ProcessLogFiles(
   PredecessorWALInfo predecessor_wal_info;
 
   for (auto wal_number : wal_numbers) {
+    SequenceNumber prev_next_sequence = *next_sequence;
     if (status.ok()) {
       status = ProcessLogFile(
           wal_number, min_wal_number, is_retry, read_only, job_id,
           next_sequence, &stop_replay_for_corruption,
           &stop_replay_by_wal_filter, &corrupted_wal_number,
           corrupted_wal_found, version_edits, &flushed, predecessor_wal_info);
     }
+    if (status.ok()) {
+      status = CheckSeqnoNotSetBackDuringRecovery(prev_next_sequence,
```

Comment:
why do we need this check in addition to ProcessLogRecord()? Is ProcessLogRecord() not the only place we can update next_sequence?? Other same-level functions as ProcessLogRecord() should hold a pointer to constant.

Reply:
Yeah, this is to detect software bugs which can happen in unexpected ways. Like for example if a bug is introduced in `ProcessLogFile` (not in ProcessLogRecord) that tampers this sequence number. This check can potentially help with that.

Reply:
Oh i see  ".... between replaying of WAL files ..." in the comment refers to this. I thought it was between "replaying of WAL files" AND "replaying of WriteBatches". 

Would you mind fixing the comment be ... between replaying of WAL files and between replaying of WriteBatches? That should clear any confusion thanks!